File: LeetCode/2997. Minimum Number of Operations to Make Array XOR Equal to K/2997.py
class Solution(object):
    def minOperations(self, nums, k):
        final_xor = 0
        # Dizideki tüm tamsayıların XOR'u.
        for n in nums:
            final_xor ^= n
        
        count = 0
        # k ve final_xor sıfır olana kadar döngüyü sürdür.
        while k > 0 or final_xor > 0:
            # k ve final_xor'un en sağdaki bitlerini karşılaştır.
            if k & 1 != final_xor & 1:
                count += 1
            
            # Her iki tamsayıdan da son bit kaldırılır.
            k >>= 1
            final_xor >>= 1
        
        return count

File: LeetCode/2997. Minimum Number of Operations to Make Array XOR Equal to K/2997.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int finalXor = 0;
        // Dizideki tüm tamsayıların XOR'u.
        for (int n : nums) {
            finalXor ^= n;
        }
        
        int count = 0;
        // k ve finalXor sıfır olana kadar döngüyü sürdür.
        while (k > 0 || finalXor > 0) {
            // k ve finalXor'un en sağdaki bitlerini karşılaştır.
            if ((k & 1) != (finalXor & 1)) {
                count++;
            }
            
            // Her iki tamsayıdan da son bit kaldırılır.
            k >>= 1;
            finalXor >>= 1;
        }
        
        return count;
    }
}
